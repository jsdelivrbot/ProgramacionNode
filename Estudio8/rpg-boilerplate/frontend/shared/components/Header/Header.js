import React from 'react'
import {createFragmentContainer} from 'react-relay'
import {graphql} from 'relay-runtime'
import createReactClass from 'create-react-class'
import AppBar from 'material-ui/AppBar'
import Toolbar from 'material-ui/Toolbar'
import { equals } from 'ramda'
import Typography from 'material-ui/Typography'
import IconButton from 'material-ui/IconButton'
import MenuIcon from 'material-ui-icons/Menu'
import Fade from 'material-ui/transitions/Fade'
import { duration } from 'material-ui/styles/transitions'
import classnames from 'classnames'

import './Header.scss'

const Header = createReactClass({
  render () {
    return (
      <div className={classnames('header', this.props.className)}>
        <AppBar position='fixed' className='header-app-bar' color='default'>
          <Toolbar className='header-toolbar'>
            <div className='header-container'>
              RPG Boilerplate
            </div>
          </Toolbar>
        </AppBar>
      </div>
    )
  }
})

export default Header

