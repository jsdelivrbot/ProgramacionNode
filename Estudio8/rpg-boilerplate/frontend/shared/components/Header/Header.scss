@import '../../theme/variables.scss';
@import '../../theme/mixins.scss';

.header {
  .header-app-bar {
    .header-toolbar {
      .header-container {
        height: $mobileHeaderHeight;

        @extend .flex-block;
        justify-content: space-between;
        align-items: center;
        overflow: hidden;

        .header-logo {
          display: none;
        }

        .header-logo-mobile {
          display: block;
          line-height: 0;
          margin: auto $mobileBlockSize;
          width: $mobileBlockSize*10;
        }

        .header-title {
          display: flex;
        }
        
        .header-mobile-search-bar {
          display: flex;
        }

        .header-theme-switcher {
          display: none;
        }

        .header-avatar {
          display: none;
        }
      }
    }
  }

  .hidden {
    display: none;
  }
}

@media screen and (min-width: $smallBreakPoint) {
  .header {
    .header-app-bar {
      .header-toolbar {
        .header-container {
          height: $desktopBlockSize*4;

          .header-logo-mobile {
            display: none;
          }

          .header-logo {
            display: block;
            margin: auto $desktopBlockSize;
            width: $desktopBlockSize*10;
          }

          .header-title {
            display: none;
          }

          .header-avatar {
            display: flex;
          }

          .header-theme-switcher {
            display: flex;
          }

          .header-mobile-search-bar {
            display: none;
          }
        }
      }
    }
  }
}
