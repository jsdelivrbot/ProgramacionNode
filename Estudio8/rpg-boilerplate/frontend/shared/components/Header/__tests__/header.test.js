/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Header from '../Header'

describe('A </Header>', function () {
  it('should be selectable by class "header"', function () {
    expect(shallow(<Header />).is('.header')).toBe(true)
  })

  it('renders correctly', () => {
    expect(toJson(shallow(<Header />))).toMatchSnapshot()
  })

  it('shows the title when fade in animation ends', () => {
    const header = shallow(<Header />)
    expect(header.state().titleShowing).toBe(false)
    header.find('.header-fade').props().onEnter()
    expect(header.state().titleShowing).toBe(true)
  })

  it('hides the title when fade out animation ends', () => {
    const header = shallow(<Header />)
    header.setState({ titleShowing: true })
    header.find('.header-fade').props().onExited()
    expect(header.state().titleShowing).toBe(false)
  })

  it('updated the state searchOpened to be true when mobile search view is opened', () => {
    const header = shallow(<Header />)
    expect(header.state().searchOpened).toBe(false)
    header.find('.header-mobile-search-bar').props().onSearchOpening()
    expect(header.state().searchOpened).toBe(true)
  })

  it('updated the state searchOpened to be false when mobile search view is closed', () => {
    const header = shallow(<Header />)
    header.setState({ searchOpened: true })
    header.find('.header-mobile-search-bar').props().onSearchClosing()
    expect(header.state().searchOpened).toBe(false)
  })
})
