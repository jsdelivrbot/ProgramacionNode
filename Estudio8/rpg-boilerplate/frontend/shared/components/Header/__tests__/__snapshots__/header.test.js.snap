exports[`A </Header> renders correctly 1`] = `
<div
  className="header">
  <withStyles(AppBar)
    className="header-app-bar"
    color="default"
    position="fixed">
    <withStyles(Toolbar)
      className="header-toolbar">
      <withStyles(IconButton)
        aria-label="Menu">
        <pure(Menu) />
      </withStyles(IconButton)>
      <div
        className="header-container">
        <withTheme(Logo)
          className="header-logo" />
        <withTheme(Fade)
          className="header-fade"
          enterTransitionDuration={250}
          in={true}
          leaveTransitionDuration={150}
          onEnter={[Function]}
          onExited={[Function]}>
          <withStyles(Typography)
            className="hidden"
            color="inherit"
            type="title">
            Title
          </withStyles(Typography)>
        </withTheme(Fade)>
        <MobileSearchBar
          className="header-mobile-search-bar"
          onSearchClosing={[Function]}
          onSearchOpening={[Function]} />
        <withTheme(SearchBar)
          className="header-search-bar" />
        <ThemeSwitcher
          className="header-theme-switcher" />
        <Avatar
          className="header-avatar" />
      </div>
    </withStyles(Toolbar)>
  </withStyles(AppBar)>
</div>
`;
